plugins/ecs-bridge/engine/engine.go

```diff
@@ -217,38 +220,59 @@ func (engine *engine) ConfigureContainerVethInterface(netnsName string, result *
 
 // ConfigureBridge configures the IP address of the bridge if needed
 func (engine *engine) ConfigureBridge(result *current.Result, bridge *netlink.Bridge) error {
-	addrs, err := engine.netLink.AddrList(bridge, syscall.AF_INET)
-	if err != nil && err != syscall.ENOENT {
-		return errors.Wrapf(err,
-			"bridge configure: unable to list addresses for bridge %s",
-			bridge.Attrs().Name)
-	}
-
 	resultBridgeNetwork := &net.IPNet{
 		IP:   result.IPs[0].Gateway,
 		Mask: result.IPs[0].Address.Mask,
 	}
 	resultBridgeCIDR := resultBridgeNetwork.String()
-	if len(addrs) > 0 {
-		for _, addr := range addrs {
-			if addr.IPNet.String() == resultBridgeCIDR {
-				return nil
-			}
-		}
-		return errors.New("bridge configure: mismatch in bridge ip address")
+
+	exists, err := engine.bridgeNetworkExists(bridge, resultBridgeCIDR)
+	if err != nil {
+		return err
+	}
+
+	if exists {
+		return nil
 	}
 
 	bridgeAddr := &netlink.Addr{
 		IPNet: resultBridgeNetwork,
 	}
 	err = engine.netLink.AddrAdd(bridge, bridgeAddr)
-	if err != nil {
-		return errors.Wrapf(err,
-			"bridge configure: unable to assign ip address to bridge %s",
+	if err == nil {
+		return nil
+	}
+
+	// if we fail to assign ip address, check bridge network exists or not again, because the error may be caused by
+	// someone else assigned the address right before we assigned it
+	exists, _ = engine.bridgeNetworkExists(bridge, resultBridgeCIDR)
+	if exists {
+		return nil
+	}
+
+	return errors.Wrapf(err,
+		"bridge configure: unable to assign ip address to bridge %s",
+		bridge.Attrs().Name)
+}
+
+func (engine *engine) bridgeNetworkExists(bridge *netlink.Bridge, bridgeNetworkCIDR string) (bool, error) {
```

Comment:
The way this method is written, you're always returning `nil` error value with `true` and a `non-nil` `error` value with `false`. Since you're ignoring the `error` returned anyway, you might as well not return it and just make this function return a `bool`. You can log the errors instead of returning and ignoring them to provide the visibility in log files.

Reply:
the function can return both (false, nil) and (false, err). the error from line 229 is not ignored which keeps the existing behavior before assigning ip address. the change is to add another address check after assigning the address

Reply:
Right, but you're ignoring the error returned anyway. So, that information is being lost. It's better to at least log it within the method and make it return just `bool` instead.

Reply:
ok, updated